autoload/vsnip/session/snippet.vim

```diff
@@ -188,9 +189,29 @@ function! s:Snippet.sync() abort
 
       let self.found_final_tabstop = self.found_final_tabstop || a:node.id == s:max_tabstop
     elseif a:node.type ==# 'variable'
-      let l:index = index(a:parent.children, a:node)
-      call remove(a:parent.children, l:index)
-      call insert(a:parent.children, vsnip#session#snippet#node#create_text(a:node.text()), l:index)
+      " variable placeholder
+      if a:node.unknown
+        let a:node.type = 'placeholder'
```

Comment:
@hrsh7th 
I moved variable placeholder logic to `s:Snippet.sync`. It works well, but looks bad...
Please review it. Thank you.

Reply:
It almost looks good.
I will try it on my machine. Please wait.

Reply:
This method is fat and a bit ugly.
But it was implemented on the correct position for me.

We can refactor it more small method in `snippet.vim` so it has no problem at the moment.

Reply:
Ok! Thank you for the review.